pkg/acsengine/params_k8s.go

```diff
@@ -259,14 +259,18 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 				CloudProviderRateLimitBucket: kubernetesConfig.CloudProviderRateLimitBucket,
 			})
 			addValue(parametersMap, "kubeClusterCidr", kubernetesConfig.ClusterSubnet)
-			if properties.OrchestratorProfile.IsAzureCNI() {
-				if properties.MasterProfile != nil && properties.MasterProfile.IsCustomVNET() {
-					addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.MasterProfile.VnetCidr)
+			if !properties.IsHostedMasterProfile() {
+				if properties.OrchestratorProfile.IsAzureCNI() {
+					if properties.MasterProfile != nil && properties.MasterProfile.IsCustomVNET() {
+						addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.MasterProfile.VnetCidr)
+					} else {
+						addValue(parametersMap, "kubernetesNonMasqueradeCidr", DefaultVNETCIDR)
+					}
 				} else {
-					addValue(parametersMap, "kubernetesNonMasqueradeCidr", DefaultVNETCIDR)
+					addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
 				}
 			} else {
-				addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
+				addValue(parametersMap, "kubernetesNonMasqueradeCidr", kubernetesConfig.KubeletConfig["--non-masquerade-cidr"])
```

Comment:
Where is `kubernetesNonMasqueradeCidr` used for hosted masters? Looking at the diff for https://github.com/Azure/acs-engine/pull/3916 it seems to only be used for masters